src/arrayvec.rs

```diff
@@ -1002,6 +1002,42 @@ impl<T, const CAP: usize> Extend<T> for ArrayVec<T, CAP> {
     }
 }
 
+/// Perform the indexing (`container[index]`) operation.
+///
+/// # Panics
+/// Panics if index is out of bounds.
+impl<I, T, const CAP: usize> Index<I> for ArrayVec<T, CAP>
+where
+    I: slice::SliceIndex<[T]>,
+{
+    type Output = <I as slice::SliceIndex<[T]>>::Output;
+
+    /// Performs the indexing (`container[index]`) operation.
+    ///
+    /// # Panics
+    /// Panics if index is out of bounds.
+    fn index(&self, index: I) -> &Self::Output {
+        self.deref().index(index)
```

Comment:
Now I guess this is a matter of taste, but IMO we don't call operator methods explicitly, we just use the relevant operator. The operator for deref is `*` and index is, you know, etc.. :slightly_smiling_face: 

```suggestion
        (**self)[index]
```

Reply:
Usually I also prefer the syntactic sugar, but when implementing `index` I always use `index` explicitly. Mainly a matter of preference. Changed it now. (The generated mir seems nearly identical. The new version uses only 1 additional local and assignment, so I don't expect a difference in runtime performance.)